server.js

```diff
@@ -13,6 +17,96 @@ app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
+app.post("/bookings", function (request, response) {
+  
+  let id = bookings.length;
+  const {
+    roomId,
+    title,
+    firstName,
+    surname,
+    email,
+    checkInDate,
+    checkOutDate,
+  } = request.body;
+  let newBooking = {
+    id,
+    roomId,
+    title,
+    firstName,
+    surname,
+    email,
+    checkInDate,
+    checkOutDate,
+  };
+  bookings.push(newBooking);
+  response.send(bookings);
+  if (
+    newBooking.roomId &&
+    newBooking.roomId > 0 &&
+    newBooking.title &&
+    newBooking.title.length > 0 &&
+    newBooking.firstName &&
+    newBooking.firstName.length > 0 &&
+    newBooking.surname &&
+    newBooking.surname.length > 0 &&
+    newBooking.email &&
+    newBooking.email.length > 0 &&
+    newBooking.checkInDate &&
+    newBooking.checkInDate.length > 0 &&
+    newBooking.checkOutDate &&
+    checkOutDate.length > 0
```

Comment:
This validation is good, however you're doing it too late; at this point you've already `bookings.push`ed whereas you actualyl want to validate _before_ adding the booking to the list, otherwise your `bookings` var could be full of invalid bookings.